### DISCUSSION TITLE
Option to disable Clock Security System (CSS) in setup

### DISCUSSION THREAD

**Discussion by abhaysbharadwaj** [stm32duino/Arduino_Core_STM32#2378] [Q&A]
Board: STM32WB5MMG (Custom PCB)
Issue: If CSS is enabled, faulty clock startup is resulting in clock getting disabled by CSS and device not waking up from sleep.

This issue is observed with STM32CubeIDE as well as Arduino. Disabling CSS in the clocktree of cube MX and re-generating the code is solving the issue with CubeIDE.

How can we disable CSS in Arduino?

Note: Issue raised with STM team. Since STM32WB5MMG is a SIP module, we have no control on the load caps of the XTALS to do anything. This seems to be CSS disabling the clock due to faulty startup. Disabling CSS ensures that the clock signals continue and sleep works.

**Comment by fpistm** [comment]
Hi @abhaysbharadwaj 
you can try to call HAL or LL to do that.
Ex: `HAL_RCCEx_DisableLSECSS()`

Anyway, I don't think CSS is enabled by default. Check your `SystemClock_Config(void)`.
